data_model/src/events/data/filters.rs

```diff
@@ -94,50 +94,60 @@ impl Filter for EntityFilter {
     }
 }
 
-#[derive(
-    Clone,
-    PartialOrd,
-    Ord,
-    PartialEq,
-    Eq,
-    Debug,
-    Decode,
-    Encode,
-    Deserialize,
-    Serialize,
-    IntoSchema,
-    Hash,
-)]
-/// Filter for identifiers
-///
-/// Passes id trough filter, if it equals to the one provided in construction
-pub struct IdFilter<Id: Eq>(Id);
+#[derive(Clone, PartialOrd, Ord, Eq, Debug, Decode, Encode, Serialize, IntoSchema)]
+/// Filter that accepts an data event whose origin matches what this filter specifies.
+pub struct OriginFilter<T: Origin>(<T::Origin as Identifiable>::Id);
 
-impl<Id: Eq> IdFilter<Id> {
-    /// Construct new `IdFilter`
-    pub fn new(id: Id) -> Self {
-        Self(id)
+impl<T: Origin> OriginFilter<T> {
+    /// Construct [`OriginFilter`].
+    pub fn new(origin_id: <T::Origin as Identifiable>::Id) -> Self {
+        Self(origin_id)
     }
 
-    /// Get `id`
-    pub fn id(&self) -> &Id {
+    /// Get the id of the origin of the data event that this filter accepts.
+    pub fn origin_id(&self) -> &<T::Origin as Identifiable>::Id {
         &self.0
     }
 }
 
-impl<Id: Eq> Filter for IdFilter<Id> {
-    type EventType = Id;
+impl<T: Origin> Filter for OriginFilter<T> {
+    type EventType = T;
+
+    fn matches(&self, event: &T) -> bool {
+        event.origin_id() == self.origin_id()
+    }
+}
+
+impl<T: Origin> PartialEq for OriginFilter<T> {
+    fn eq(&self, other: &Self) -> bool {
+        self.0 == other.0
+    }
+}
+
+impl<T: Origin> core::hash::Hash for OriginFilter<T> {
+    fn hash<H: core::hash::Hasher>(&self, state: &mut H) {
+        self.0.hash(state);
+    }
+}
 
-    fn matches(&self, id: &Id) -> bool {
-        id == &self.0
+impl<'de, T: Origin> Deserialize<'de> for OriginFilter<T>
+where
+    <<T as Origin>::Origin as Identifiable>::Id: Deserialize<'de>,
+{
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let origin_id = <T::Origin as Identifiable>::Id::deserialize(deserializer)?;
+        Ok(Self::new(origin_id))
     }
 }
```

Comment:
This is an alternative to forcing `Identifiable::Id` to be `Deserialize`, otherwise it would bring larger change

Reply:
Do you mean this in terms of the amount of generated code, or would it require manual impls?

Reply:
I mean manual changes caused by the lifetime parameter. The idea is to minimize changes so as not to disturb #2476